activerecord/lib/active_record/associations.rb

```diff
@@ -1176,8 +1189,6 @@ def association_instance_set(name, association)
       # it returns the association rather than the records which have been deleted.
       #
       # === What gets deleted?
-      #
-      # There is a potential pitfall here: #has_and_belongs_to_many and #has_many <tt>:through</tt>
```

Comment:
Was this removed accidentally?

Reply:
I believe so, I will put this back. Good catch.